dbt_copilot_helper/commands/dns.py

```diff
@@ -247,140 +237,200 @@ def add_records(client, records, subdom_id, action):
     return response["ChangeInfo"]["Status"]
 
 
-def check_for_records(client, parent_id, subdom, subdom_id):
-    records_to_add = []
-    response = client.list_resource_record_sets(
-        HostedZoneId=parent_id,
-    )
+def copy_records_from_parent_to_subdomain(client, parent_id, subdom, subdom_id):
+    resource_record_sets = client.list_resource_record_sets(HostedZoneId=parent_id)[
+        "ResourceRecordSets"
+    ]
 
-    for records in response["ResourceRecordSets"]:
+    for records in resource_record_sets:
         if subdom in records["Name"]:
             click.secho(records["Name"] + " found", fg="green")
-            records_to_add.append(records)
             add_records(client, records, subdom_id, "UPSERT")
-    return True
 
 
-def create_hosted_zone(client, domain, start_domain, base_len):
-    parts = domain.split(".")
+def create_hosted_zones(client, base_domain, subdomain):
+    domains_to_create = get_required_subdomains(base_domain, subdomain)
+    zone_ids = {}
 
-    # We only want to create domains max 2 deep so remove all sub domains in excess
-    parts_to_remove = len(parts) - base_len - MAX_DOMAIN_DEPTH
-    domain_to_create = parts[parts_to_remove:]
-
-    # Walk back domain name
-    for x in reversed(range(len(domain_to_create) - (len(start_domain.split(".")) - 1))):
-        subdom = ".".join(domain_to_create[(x):]) + "."
+    for dom in domains_to_create:
```

Comment:
I reckon I would favour `domain` or `subdomain` instead of `dom`.

Reply:
Yeah, I don't like dom really, but domain shadows the domain() function at the top level. Which is only really an irritation as the IDE complains. I'll rename it to something sensible.